app/lib/screens/wallets/contacts.dart

```diff
@@ -38,9 +38,21 @@ class _ContactsScreenState extends State<ContactsScreen> {
   }
 
   _loadFavouriteContacts() async {
-    myPkidContacts = await getPkidContacts();
-    myPkidContacts =
-        myPkidContacts.where((c) => c.type == widget.chainType).toList();
+    final allContacts = await getPkidContacts();
+
+    final seen = <String>{};
+    final filtered = <PkidContact>[];
+
+    for (final c in allContacts) {
+      if (c.type != widget.chainType) continue;
+
+      final key = '${c.address}_${c.type}';
+      if (seen.add(key)) {
+        filtered.add(c);
+      }
+    }
+
+    myPkidContacts = filtered;
     setState(() {});
   }
 

```

Comment:
I don't think this is necessary. Should be filtered once [this](https://github.com/threefoldtech/threefold_connect/pull/986) is merged.